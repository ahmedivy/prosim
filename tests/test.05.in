32 5 8
Proc1 1 1 1
HALT

Proc1 1 1 1
HALT

Proc1 1 1 1
HALT

Proc1 1 1 1
HALT

Proc1 1 1 2
HALT

Proc1 1 1 2
HALT

Proc1 1 1 2
HALT

Proc1 1 1 2
HALT

Proc1 1 1 3
HALT

Proc1 1 1 3
HALT

Proc1 1 1 3
HALT

Proc1 1 1 3
HALT

Proc1 1 1 4
HALT

Proc1 1 1 4
HALT

Proc1 1 1 4
HALT

Proc1 1 1 4
HALT

Proc1 1 1 5
HALT

Proc1 1 1 5
HALT

Proc1 1 1 5
HALT

Proc1 1 1 5
HALT

Proc1 1 1 6
HALT

Proc1 1 1 6
HALT

Proc1 1 1 6
HALT

Proc1 1 1 6
HALT

Proc1 1 1 7
HALT

Proc1 1 1 7
HALT

Proc1 1 1 7
HALT

Proc1 1 1 7
HALT

Proc1 1 1 8
HALT

Proc1 1 1 8
HALT

Proc1 1 1 8
HALT

Proc1 1 1 8
HALT
