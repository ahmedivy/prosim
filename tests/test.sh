#!/bin/sh
export LC_ALL=C

echo ======================================================
echo ====================== TEST $1 =======================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.raw; then 
  cat tests/test.$1.raw | sort > tests/test.$1.out
  if diff -b tests/test.$1.out tests/test.$1.expected > /dev/null; then
    if grep "IS_CONCURRENT" tests/test.$1.cfg > /dev/null; then
      for x in {0..100}; do
        if diff tests/test.$1.raw tests/test.$1.out > /dev/null; then
          if [ $x -eq 100 ]; then
            echo FAILED: Output is correct, but no concurrency is apparent
            echo Threads appear to be executed in sequential order
            exit 1
          else 
            echo RETRYING: Output is correct, but no concurrency is apparent
            timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.raw
          fi
        else 
          break
        fi
      done
      #if diff tests/test.$1.raw tests/test.$1.out > /dev/null; then
      #fi
    fi
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ______________Your_output______________ ____________Expected_Output_____________
   #echo +++++++++++++++++++++++++++++++++++++++ ++++++++++++++++++++++++++++++++++++++++
    diff -b -y -W 80 tests/test.$1.out tests/test.$1.expected
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
