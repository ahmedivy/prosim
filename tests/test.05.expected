[01] 00000: process 1 finished
[01] 00000: process 1 new
[01] 00000: process 2 finished
[01] 00000: process 2 new
[01] 00000: process 3 finished
[01] 00000: process 3 new
[01] 00000: process 4 finished
[01] 00000: process 4 new
[02] 00000: process 1 finished
[02] 00000: process 1 new
[02] 00000: process 2 finished
[02] 00000: process 2 new
[02] 00000: process 3 finished
[02] 00000: process 3 new
[02] 00000: process 4 finished
[02] 00000: process 4 new
[03] 00000: process 1 finished
[03] 00000: process 1 new
[03] 00000: process 2 finished
[03] 00000: process 2 new
[03] 00000: process 3 finished
[03] 00000: process 3 new
[03] 00000: process 4 finished
[03] 00000: process 4 new
[04] 00000: process 1 finished
[04] 00000: process 1 new
[04] 00000: process 2 finished
[04] 00000: process 2 new
[04] 00000: process 3 finished
[04] 00000: process 3 new
[04] 00000: process 4 finished
[04] 00000: process 4 new
[05] 00000: process 1 finished
[05] 00000: process 1 new
[05] 00000: process 2 finished
[05] 00000: process 2 new
[05] 00000: process 3 finished
[05] 00000: process 3 new
[05] 00000: process 4 finished
[05] 00000: process 4 new
[06] 00000: process 1 finished
[06] 00000: process 1 new
[06] 00000: process 2 finished
[06] 00000: process 2 new
[06] 00000: process 3 finished
[06] 00000: process 3 new
[06] 00000: process 4 finished
[06] 00000: process 4 new
[07] 00000: process 1 finished
[07] 00000: process 1 new
[07] 00000: process 2 finished
[07] 00000: process 2 new
[07] 00000: process 3 finished
[07] 00000: process 3 new
[07] 00000: process 4 finished
[07] 00000: process 4 new
[08] 00000: process 1 finished
[08] 00000: process 1 new
[08] 00000: process 2 finished
[08] 00000: process 2 new
[08] 00000: process 3 finished
[08] 00000: process 3 new
[08] 00000: process 4 finished
[08] 00000: process 4 new
| 00000 | Proc 01.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 01.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 01.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 01.04 | Run 0, Block 0, Wait 0
| 00000 | Proc 02.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 02.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 02.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 02.04 | Run 0, Block 0, Wait 0
| 00000 | Proc 03.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 03.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 03.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 03.04 | Run 0, Block 0, Wait 0
| 00000 | Proc 04.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 04.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 04.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 04.04 | Run 0, Block 0, Wait 0
| 00000 | Proc 05.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 05.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 05.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 05.04 | Run 0, Block 0, Wait 0
| 00000 | Proc 06.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 06.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 06.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 06.04 | Run 0, Block 0, Wait 0
| 00000 | Proc 07.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 07.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 07.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 07.04 | Run 0, Block 0, Wait 0
| 00000 | Proc 08.01 | Run 0, Block 0, Wait 0
| 00000 | Proc 08.02 | Run 0, Block 0, Wait 0
| 00000 | Proc 08.03 | Run 0, Block 0, Wait 0
| 00000 | Proc 08.04 | Run 0, Block 0, Wait 0
